modules/hitl2/src/views/full/App.tsx

```diff
@@ -0,0 +1,130 @@
+import { lang, MainLayout, toast } from 'botpress/shared'
+import _ from 'lodash'
+import React, { useContext, useEffect, useState } from 'react'
+
+import { SocketMessageType } from './../../types'
+import { Context, Store } from './app/Store'
+import style from './style.scss'
+import { Api, castEscalation } from './Api'
+import AgentProfile from './Components/AgentProfile'
+import Conversation from './Components/Conversation'
+import EscalationList from './Components/EscalationList'
+
+const App = ({ bp }) => {
+  const api = Api(bp)
+
+  const { state, dispatch } = useContext(Context)
+
+  const [loading, setLoading] = useState(true)
+
+  function handleMessage(message: SocketMessageType) {
+    switch (message.resource) {
+      case 'agent':
+        return dispatch({ type: 'setAgent', payload: message })
+      case 'escalation':
+        return dispatch({
+          type: 'setEscalation',
+          payload: _.thru(message, () => {
+            message.payload = castEscalation(message.payload)
+            return message
+          })
+        })
+      default:
+        return
+    }
+  }
+
+  async function getCurrentAgent() {
+    try {
+      const agent = await api.getCurrentAgent()
+      dispatch({ type: 'setCurrentAgent', payload: agent })
+    } catch (error) {
+      dispatch({ type: 'setError', payload: error })
+    }
+  }
+
+  async function getAgents() {
+    try {
+      const agents = await api.getAgents()
+      dispatch({ type: 'setAgents', payload: agents })
+    } catch (error) {
+      dispatch({ type: 'setError', payload: error })
+    }
+  }
+
+  async function getEscalations() {
+    try {
+      const escalations = await api.getEscalations()
+      dispatch({ type: 'setEscalations', payload: escalations })
+    } catch (error) {
+      dispatch({ type: 'setError', payload: error })
+    }
+  }
+
+  async function toggleOnline(online: boolean) {
+    try {
+      let agent
+      if (online) {
+        agent = await api.setOnline()
+      } else {
+        agent = await api.setOffline()
+      }
```

Comment:
Can be simplified: ` const agent = online ? await api.setOnline() : await api.setOffline()`